misk/src/main/kotlin/misk/Deadline.kt

```diff
@@ -0,0 +1,103 @@
+package misk
+
+import misk.exceptions.BadRequestException
+import misk.scope.ActionScoped
+import misk.scope.ActionScopedProvider
+import misk.web.HttpCall
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import javax.inject.Inject
+
+/**
+ * Deadline for an action. A deadline can be overridden, with the caveat that the overridden deadline
+ * does not propagate to other threads.
+ *
+ * All actions have [ActionDeadline]s available even if there is initially no deadline.
```

Comment:
As a follow up I’d love our default deadline to be say, 10 seconds. You can specify a longer deadline, but the default is not infinite.